### PR TITLE
Modifications to `annotate_freq` to improve memory use

### PR DESCRIPTION
**Pull request by jkgoodrich**

### CODE REVIEW

**Review by mike-w-wilson — COMMENTED** [review]
I really like this. The logic all looks great and this is way cleaner on top of improving performance. I just have a couple suggestions and questions.

File: gnomad/utils/annotations.py [hunk 1]
```diff
@@ -1646,39 +1758,29 @@ def _agg_by_group(
...
-            lambda cs: cs.annotate(
-                AC=cs.AC[1],
-                # TODO: This is NA in case AC and AN are 0 -- should we set it to 0?
-                AF=cs.AF[1],
-                homozygote_count=cs.homozygote_count[1],
-            )
+    freq_expr = freq_expr.map(
+        lambda cs: cs.annotate(
+            AC=cs.AC[1],
+            # TODO: This is NA in case AC and AN are 0 -- should we set it to 0?
```
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad/utils/annotations.py:1775]: Sure, can you initiate the conversation?

File: gnomad/utils/annotations.py [hunk 2]
```diff
@@ -1646,39 +1762,29 @@ def _agg_by_group(
...
```
Comment (mike-w-wilson) [inline gnomad/utils/annotations.py:1779]:
```suggestion
```

**Review by mike-w-wilson — APPROVED** [review]
LGTM!

File: gnomad/utils/annotations.py (hunk 2 above)
Comment (mike-w-wilson) [reply gnomad/utils/annotations.py]: Konrad wants NAs

**Pull request merged**